### PR TITLE
ktlo-0001 - Refactor update quality method

### PR DESCRIPTION
**Pull request by willunderwood**
ktlo-0001 - some_jira_instance_link.com/ktlo-0001

**Desc**
Refactor update quality method and increase use case and edge case test coverage.

Considerations
 - Wanted to try to keep it as a single module and only add new code to the one class so there is no code bloat etc

Limitations
 - The `_is_<item_type>` identifier methods are fine for now, but could become problematic if many new items types are added. But this can be resolved in the future by further re-factoring making a single `_is_<item_type>` method that is dymanic enough to work for all, (maybe by moving item type definitions and their rules to a config file or database, and new item types and their rules added directly to the config or db by users / devs)

### CODE REVIEW

File: python/gilded_rose.py [hunk 1]
```diff
@@ -1,39 +1,96 @@
...
+    STANDARD_INCREMENT = 1
 
     def __init__(self, items):
         self.items = items
 
+    @staticmethod
+    def _is_static_item(name: str) -> bool:
+        static_items = [
+            'Sulfuras, Hand of Ragnaros',
+        ]
```
Comment (benjamincerigo) [inline python/gilded_rose.py:15]: Maybe these sort of things could go into a `constants.py` module?

File: python/test_gilded_rose.py [hunk 2]
```diff
@@ -9,8 +9,92 @@ def test_foo(self):
...
+        items = [Item("Conjured", 10, 30)]
+        gilded_rose = GildedRose(items)
+        gilded_rose.update_quality()
+        self.assertEquals(28, items[0].quality)
+
+    def test_conjured_item_quality_increase_double_when_sell_in_negative(self):
+        items = [Item("Conjured", 0, 30)]
+        gilded_rose = GildedRose(items)
+        gilded_rose.update_quality()
+        self.assertEquals(26, items[0].quality)
```
Comment (benjamincerigo) [inline python/test_gilded_rose.py:96]: The tests file is a bit long maybe it is worth breaking that apart?

**Review by benjamincerigo — COMMENTED** [review]
Looks great!
